index.js

```diff
@@ -1,51 +1,166 @@
+// create event listener. on click
+deleteButton.addEventListener('click', () => {
+    // create a variable that looks for the select list (id = colorsList)
+    let removeItem = document.getElementById('colorsList');
+    // and remove the selected item 
+    removeItem.remove(removeItem.selectedIndex);
+})
 
 //------------------------Question 3 ---------------------------
-//Now lets mix some maths and programming. In the function below (calculate_sphere) write a JavaScript program to calculate the volume and surface area of a sphere.
+//Now lets mix some maths and programming. In the function below (calculate_sphere)
+// write a JavaScript program to calculate the volume and surface area of a sphere.
 //take the radius from the input value element with the 'id=radius' in the index.html file
 //And display the results in the elements with the 'id = volume' and 'id = area' respectively
 //hint: the volumen of a sphere is ((4/3) × π × r^3) and the surface area is (4 × π × r^2)
 
 
-const calculate_sphere = () =>{
+// get the input for "volume"
+const volume = document.querySelector('#volume');
+const area = document.querySelector('#area');
+const a3 = document.getElementById('a-3');
 
- }
+const calculate_sphere = () =>{
+  // get input of radius
+  let radius = document.querySelector('#radius').value;
+  let radiusInput = Number.parseInt(radius);
+  
+  // create and perform math and insert answer as value
+  let areaFormula = (4 * Math.PI * Math.pow(radiusInput, 2)).toFixed(2);
+  let volumeFormula = (((4/3) * Math.PI * (Math.pow(radiusInput, 3)))).toFixed(2);
+  volume.setAttribute('value', `${volumeFormula} cubic units`);
+  area.setAttribute('value', `${areaFormula} sq units`);
+  
+  // return false to prevent form default clearing behavior
+  return false;
+};
+    
+ 
 
 window.onload = document.getElementById('MyForm').onsubmit = calculate_sphere; // this execute the volume_sphere function each time the calculate (submit) button is clicked
 
 
 
 
 //------------------------Question 4 ---------------------------
-//Now in this exercise we want you to create 3 buttons, each with a click action that will hide the respective question's answer above. So if you click the "Hide Question 1" button it will hide from the DOM the answer to your first exercise. If you click it again it will show the answer. 
+//Now in this exercise we want you to create 3 buttons, 
+//each with a click action that will hide the respective question's answer above. 
+//So if you click the "Hide Question 1" button it will hide from the DOM the answer 
+//to your first exercise. If you click it again it will show the answer. 
 
 
 //resolve // QUESTION 4 here
+
+// this can be done in a function to create a button for each question
+// only need it for three questions
+// sooooo...
+const a4 = document.getElementById('a-4');
+
+// name the function. the only argument i need is the id of the section i want to hide.
+function createHideButton (section) {
```

Comment:
If we are using arrow functions (line 100), it would be better to use them throughout all the code.